mitiq/calibration/calibrator.py

```diff
@@ -210,12 +222,22 @@ def run(self, log: bool = False) -> None:
                 self.cirq_executor, bitstring_to_measure
             )
             noisy_value = expval_executor.evaluate(circuit)[0]
-            for strategy in self.strategies:
+            for count, strategy in enumerate(self.strategies):
                 with warnings.catch_warnings():
                     warnings.simplefilter("ignore", UserWarning)
                     mitigated_value = strategy.mitigation_function(
                         circuit, expval_executor
                     )
+                if (
+                    log
+                    and count
+                    and strategy.technique
+                    is not self.strategies[count - 1].technique
+                ):
+                    if strategy.technique is MitigationTechnique.ZNE:
+                        print(TABLE_HEADER_STR_ZNE)
+                    elif strategy.technique is MitigationTechnique.PEC:
+                        print(TABLE_HEADER_STR_PEC)
```

Comment:
I think this is necessary for the moment, but I don't love how we now have to print table headers in two places (in the code). I feel like it's going to need to be refactored at some point to make all this work a little cleaner, but I'm not sure what that is now. Any thoughts?

Reply:
I tried removing the `if count` condition and the code for initial printing of the header, which I thought would be a more succinct equivalent, but then the first header wasn't printed so I had to revert back.